### ISSUE TITLE
🐛 2024.10.0 appears to have broken xterm.js (e.g. Proxmox WebUI) when served over CF tunnel

### ISSUE THREAD

**Issue by itsthejb** [cloudflare/cloudflared#1337]
**Describe the bug**

Prior to `2024.10.0` it was possible to serve [Proxmox](https://www.proxmox.com/en/) web UI over a CF tunnel. However since `2024.10.0` the default `xterm.js` CT/VM web console is failing with `Connection failed (Error 501: )`. The `noVNC` implementation continues to work without issues.

**To Reproduce**
- Create tunnel pointing to Proxmox web UI at port 8006
- Serve tunnel using client version `2024.10.0`
- Attempt to connect to a LXC/VM console using the `>_ Shell` button
- Fails w/ `Connection failed (Error 501: )`

**Expected behavior**
- Possible to connect to the console as prior to `2024.10.0`

**Environment and versions**
 - OS: Proxmox 8.2 (latest) / Debian Bookworm
 - Architecture: x86_64
 - Version: `2024.10.0`

**Logs and errors**
- I don't currently see any clear error log entry from `journalctl` for `cloudflared` when the failure happens

**Additional context**
- ~~Would like to test a downgrade to `2024.9.1` to verify the problem exactly, but not sure how this is possible~~ Verified that the issue was introduced after `2024.9.1` by restoring to a backup w/ `2024.9.1`, where everything works as before
- In my case, `cloudflared` installed w/ `apt` in a Debian Bookworm LXC

**Comment by OdoMarTus** [comment]
I confirm. 
I have the same problem. 
However, it appeared for me about two or three weeks ago.

**Comment by itsthejb** [comment]
Confirmed that, at least in my case, the issue only began after `2024.9.1`. I reverted to a Proxmox CT backup that had `cloudflared@2024.9.1` and all is working again. This forces me to hold the package, however...

**Comment by q0Jop** [comment]
I can also confirm. I did not run into this issue until upgrading to 2024.10.0 recently.

**Comment by kmendell** [comment]
I confirm this issue as well.

**Comment by itsthejb** [comment]
@maggie44 I'll try this out later

**Comment by itsthejb** [comment]
I bisected the issue (https://github.com/cloudflare/cloudflared/pull/1340#issuecomment-2422660954) and found the breaking commit to be e2064c820f32802f58baa027903043fe0ed052e0

**Comment by ignSKRRRTT** [comment]
ISSUE #1337 🔥 🤑

**Comment by GoncaloGarcia** [comment]
Hey everyone! Thank you for letting us know of this problem and for the effort in identifying the cause of the issue. We decided to fully revert the change and will take some time to evaluate a better approach. Please let us know if you still see this issue after upgrading to 2024.10.1.

**Comment by itsthejb** [comment]
Can confirm things are working as expected in 2024.10.1

**Comment by itsthejb** [comment]
@GoncaloGarcia In spite of reverting the change, installing the new version it really feels like the interface is less reliable than previous. While the console loads, it frequently disconnects and I have to refresh the page. Do you think this could still be related to other changes in `2024.10.0`?

**Comment by OdoMarTus** [comment]
Fixed in the latest version 2024.10.1

**Comment by joliveirinha** [comment]
@itsthejb what do you mean by interface? Proxmox interface?

The code in 2024.10.0 had very minimal changes and the only one that could affect traffic was the one we reverted. so it should not be related.
Please open a new issue with more details if you can reproduce it.

**Comment by itsthejb** [comment]
@joliveirinha Yes, it's not the original issue, which is definitely fixed. I just noticed the web console stalling and losing the connection but differently. I could indeed be some different, so I just wanted to ask in case you thought there might be something else.

Sure, I'll leave the issue closed. I suspect it is something else, just wanted to ask why the topic is still warm.

Cheers

**Comment by itsthejb** [comment]
FYI the disconnection error is:

<img width="362" alt="Screenshot 2024-10-29 at 11 38 08" src="https://github.com/user-attachments/assets/8ec9d97a-f4de-474e-9520-3104b6758a8d">

**Comment by clemone210** [comment]
can confirm that with 2024.10.1 the issue is not present anymore.
